packages/web/src/components/reddit-card.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import axios from "axios";
+import RedditCardDetails from "./reddit-card-details";
+import { useState, useEffect } from "react";
+import { Card, Typography } from "antd";
+const { Title } = Typography;
+interface Props {
+  height: number;
+  width: number;
+}
+const RedditCard: React.FC<Props> = (size) => {
+  const [postData, setpostData] = useState([]);
+  const [fetched, setFetched] = useState(false);
+  const data = async () => {
+    const res = await axios.get(
+      "https://www.reddit.com/r/CompanyOfHeroes/top.json?limit=50&t=week",
+    );
+    setpostData(res?.data?.data?.children);
+    setFetched(true);
+    return postData;
+  };
+  useEffect(() => {
+    data();
+  }, [fetched]);
+
+  return (
+    <div>
+      <Card style={{ width: size.width, height: size.height, flexGrow: 1 }}>
+        <div style={{ overflow: "hidden" }}>
+          <Title level={3}>Top Reddit Posts</Title>
+          <p>r/CompanyOfHeros</p>
+          {postData
+            .filter(function (e: any) {
+              return e?.data?.link_flair_text === "CoH2";
+            })
+            .slice(0, 10)
```

Comment:
I would move the filter and slice into  the data function. In case something would trigger re-render of the component. These (filter and slice) would be run too. When we move it , it won't be run because it will be already saved in the state.